flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/DataType.java

```diff
@@ -86,10 +87,29 @@ public LogicalType getLogicalType() {
         return conversionClass;
     }
 
+    /**
+     * Returns the children of this data type, if any. Returns an empty list if this data type is
+     * atomic.
+     *
+     * @return the children data types
+     */
     public abstract List<DataType> getChildren();
 
+    /**
+     * Visit this data type.
+     *
+     * @return the result of the visit
+     */
     public abstract <R> R accept(DataTypeVisitor<R> visitor);
 
+    /**
+     * Creates a {@link DataType} from this instance with internal data structures conversion
+     * classes.
```

Comment:
give an example e.g. `RowData` instead of `Row`, mention that it also updated the nested fields